sdk/cosmos/azure_data_cosmos/src/cosmos_request.rs

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::operation_context::OperationType;
+use crate::request_context::RequestContext;
+use crate::resource_context::ResourceType;
+use crate::{constants, PartitionKey};
+use azure_core::http::headers::{AsHeaders, HeaderName, HeaderValue, Headers};
+use azure_core::http::{
+    request::{options::ContentType, Request},
+    Method,
+};
+
+/// Specifies which form of authorization token should be used when signing
+/// the request. The SDK generally uses the primary key, but some operations
+/// may be signed with a resource token (e.g. restricted delegation scenarios).
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub enum AuthorizationTokenType {
+    /// Use the account's primary (or secondary) key material.
+    Primary,
+    /// Use a resource token scoped to a particular resource (fine‑grained auth).
+    Resource,
+}
+
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub struct PartitionKeyRangeIdentity {
+    pub collection_rid: String,
+    pub partition_key_range_id: String,
+}
+
+/// Internal representation of a Cosmos DB operation before it is converted
+/// into a wire-level `azure_core::http::Request`.
+///
+/// It collects operation intent (create/read/query/etc.), resource routing
+/// information, partition key, optional item-level options and flags that
+/// influence retry or gateway behaviors.
+#[derive(Clone)]
```

Comment:
Do you think we could derive `PartialEq` and `Eq` here? I think all the field types are equatable (even the headers and body) and it would make some testing and other scenarios easier.